### PR TITLE
(GEOT-4963) Added fix for ContentFeatureStore.getCount() with startIndex...

### PR DESCRIPTION
**Pull request by tbarsballe**
Fix for ContentFeatureStore.getCount() with startIndex or MaxFeatures set (https://jira.codehaus.org/browse/GEOT-4963).
Modified test cases in ContentFeatureStore to test for getCount()
Added test cases to applicible extending classes.

### CODE REVIEW

File: modules/library/data/src/main/java/org/geotools/data/store/ContentFeatureSource.java [hunk 1]
```diff
@@ -501,6 +501,21 @@ public final int getCount(Query query) throws IOException {
             }
         }
         
+        // offset
```
Comment (jodygarnett) [inline modules/library/data/src/main/java/org/geotools/data/store/ContentFeatureSource.java]: Looks fixed now (I see a check for if (count < 0 ).

**Comment by jodygarnett** [comment]
I expect the CSVDataStore test is now out of date...

File: modules/unsupported/csv/src/test/java/org/geotools/data/csv/CSVDataStoreTest.java [hunk 2]
```diff
@@ -159,13 +161,94 @@ public void testFilter() throws FileNotFoundException, IOException {
...
             SimpleFeature f = iter.next();
-            filteredCities.contains(f.getAttribute(CITY_COL));
+            assertTrue(filteredCities.contains(f.getAttribute(CITY_COL)));
             count++;
         }
         iter.close();
         assertEquals(3, count);
         assertEquals(3, matches.size());
     }
+
```
Comment (jodygarnett) [inline modules/unsupported/csv/src/test/java/org/geotools/data/csv/CSVDataStoreTest.java]: I was unable to grab these three changes as the tests for CSVDataStore have been changed/removed.

Comment (tbarsballe) [inline modules/unsupported/csv/src/test/java/org/geotools/data/csv/CSVDataStoreTest.java]: These changes have been added to the new CSVDataStoreTest in  #666

**Comment by tbarsballe** [comment]
Closed via 991e675 and #666

**Pull request closed without merging**